### ISSUE TITLE
Python type hints support

### ISSUE THREAD

**Issue by KOLANICH** [doxygen/doxygen#6462]
Now it doesn't support them and even recognizes arguments incorrectly (some arguments are not recognized)

**Comment by albert-github** [comment]
Can you please attach a, small, self contained example (source+config file in a tar or zip) that allows us to reproduce the problem?
Please also specify the doxygen version used.

And also check the other issue reports (whether or not there is already a report for this)

**Comment by KOLANICH** [comment]
```
OUTPUT_DIRECTORY  = "./doc"
EXTRACT_ALL       = YES
RECURSIVE         = YES
SOURCE_BROWSER    = YES
INLINE_SOURCES    = YES
```

```python
def a(b, c:int=2, d:float=3., e:str=None, f:float=4.):
	pass
```

**Comment by zhangzq** [comment]
I found it too.

It's very annoying. Please fix it asap. thanks.

**Comment by UwePfautsch** [comment]
I use an input filter script as workaround.
cf. [https://github.com/UwePfautsch/doxyfilter_python](url)

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.15.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
